### PR TITLE
Package manager UI Fixes

### PR DESCRIPTION
**Pull request by AlexanderSher**

### CODE REVIEW

File: src/Package/Impl/Options/R/Tools/RToolsSettingsImplementation.cs [hunk 1]
```diff
@@ -45,6 +48,16 @@ internal sealed class RToolsSettingsImplementation : IRToolsSettings {
 
         public bool MultilineHistorySelection { get; set; } = true;
 
+        public bool ShowPackageManagerDisclaimer
+        {
```
Comment (huguesv) [inline src/Package/Impl/Options/R/Tools/RToolsSettingsImplementation.cs]: formatting :)

File: src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs [hunk 2]
```diff
@@ -127,77 +134,163 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
...
+                IsLoading = false;
+            }
+
             if (_availablePackages == null) {
                 return;
             }
             _items.ReplaceWith(_availablePackages);
         }
 
+        private async Task EnsureAvailablePackagesLoadedAsync() {
```
Comment (int19h) [inline src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs]: It looks like you could switch to bg thread here already - nothing in this needs UI.

**Comment by int19h** [comment]
:+1:

File: src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs [hunk 3]
```diff
@@ -125,79 +184,180 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
...
+        private async Task ReloadInstalledAndLoadedPackagesAsync() {
+            _installedAndLoadedLock.ResetIfNotWaiting();
+            var areLoaded = await _installedAndLoadedLock.WaitAsync();
+            if (areLoaded) {
+                return;
+            }
+
+            try {
+                await LoadInstalledAndLoadedPackagesAsync();
+            } catch (RPackageManagerException ex) {
```
Comment (huguesv) [inline src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs]: ex not used, is there a compiler warning here?

**Comment by huguesv** [comment]
:+1:

File: src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs (hunk 3 above)
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs]: Yes, there is a warning, and there will be logic in the next PR :)

**Pull request merged**